packages/backend/src/models/CatalogModel/CatalogModel.ts

```diff
@@ -697,11 +698,23 @@ export class CatalogModel {
 
     async getMetricsTree(
         projectUuid: string,
+        metricNodes: CatalogMetricsTreeNode[],
     ): Promise<{ edges: CatalogMetricsTreeEdge[] }> {
-        const edges = await this.database(MetricsTreeEdgesTableName).where(
-            'project_uuid',
-            projectUuid,
-        );
+        const metricNodeEntries = metricNodes.map((n) => [n.name, n.tableName]);
+        const edges = await this.database(MetricsTreeEdgesTableName)
+            .where('project_uuid', projectUuid)
+            .andWhere(function sourceNodeWhere() {
+                void this.whereIn(
+                    ['source_metric_name', 'source_metric_table_name'],
+                    metricNodeEntries,
+                );
+            })
+            .andWhere(function targetNodeWhere() {
```

Comment:
Yep, for now since we're sure that we'll always know the nodes displayed on the canvas we're just getting the edges containing source/target included in that set (to limit the amount of things we get in the frontend)

But it should be easy to change if we decide to have a more complex feature just like the one you mentioned 👌